src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -633,6 +636,127 @@ public void Add(DocumentItem document)
             {
                 Documents.Add(document);
             }
+
+            public DynamicJsonValue Serialize()
+            {
+                var ret = new DynamicJsonValue
+                {
+                    [nameof(BuildVersionType)] = _buildType,
+                    [nameof(Documents)] = Documents.Select(DocumentItemToDJson)
+                };
+
+                if (IsRevision)
+                {
+                    ret[nameof(IsRevision)] = true;
+                }
+
+                return ret;
+            }
+
+            private DynamicJsonValue DocumentItemToDJson(DocumentItem documentItem)
+            {
+                var ret = new DynamicJsonValue
+                {
+                    [nameof(DocumentItem.Document)] = DocumentToDJson(documentItem.Document)
+                };
+
+                return ret;
+            }
+
+            private static DynamicJsonValue DocumentToDJson(Document documentItem)
+            {
+                var document = new DynamicJsonValue
+                {
+                    [nameof(Document.Id)] = documentItem.Id,
+                    [nameof(Document.Data)] = documentItem.Data,
+                    [nameof(Document.Flags)] = documentItem.Flags,
+                    [nameof(Document.Etag)] = documentItem.Etag,
+                    [nameof(Document.LastModified)] = documentItem.LastModified,
+                };
+                if (string.IsNullOrEmpty(documentItem.ChangeVector) == false)
+                {
+                    document[nameof(Document.ChangeVector)] = documentItem.ChangeVector;
+                }
+
+                return document;
+            }
+
+            public static MergedBatchPutCommand Deserialize(BlittableJsonReaderObject mergedCmdReader, DocumentDatabase database, JsonOperationContext context)
+            {
+                if (mergedCmdReader.TryGet(nameof(BuildVersionType), out BuildVersionType type) == false)
+                {
+                    throw new Exception($"Can't read {nameof(BuildVersionType)} from {nameof(MergedBatchPutCommand)}");
+                }
+
+                //Todo To check if log is necessary 
```

Comment:
Yes, but you can provide it in the same manner as usual

Reply:
done